src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvection.hpp

```diff
@@ -146,31 +148,55 @@ struct EvolutionMetavars {
   using time_stepper_tag = Tags::TimeStepper<
       tmpl::conditional_t<local_time_stepping, LtsTimeStepper, TimeStepper>>;
 
-  using observe_fields = typename system::variables_tag::tags_list;
-  using analytic_solution_fields =
-      tmpl::conditional_t<is_analytic_solution_v<initial_data>, observe_fields,
-                          tmpl::list<>>;
+  using analytic_variables_tags = typename system::variables_tag::tags_list;
+  using analytic_compute =
+      evolution::Tags::AnalyticSolutionsCompute<volume_dim,
+                                                analytic_variables_tags>;
+  using error_compute = Tags::ErrorsCompute<analytic_variables_tags>;
+  using error_tags = db::wrap_tags_in<Tags::Error, analytic_variables_tags>;
+  using observe_fields = tmpl::push_back<
+      tmpl::append<typename system::variables_tag::tags_list, error_tags,
+                   tmpl::conditional_t<
+                       use_dg_subcell,
+                       tmpl::list<evolution::dg::subcell::Tags::TciStatus>,
+                       tmpl::list<>>>,
+      tmpl::conditional_t<
+          use_dg_subcell,
+          evolution::dg::subcell::Tags::ObserverCoordinatesCompute<volume_dim,
+                                                                   Frame::Grid>,
+          domain::Tags::Coordinates<volume_dim, Frame::Grid>>,
+      tmpl::conditional_t<
+          use_dg_subcell,
+          evolution::dg::subcell::Tags::ObserverCoordinatesCompute<
+              volume_dim, Frame::Inertial>,
+          domain::Tags::Coordinates<volume_dim, Frame::Inertial>>>;
+  using non_tensor_compute_tags = tmpl::list<
+      tmpl::conditional_t<
+          use_dg_subcell,
+          evolution::dg::subcell::Tags::ObserverMeshCompute<volume_dim>,
+          ::Events::Tags::ObserverMeshCompute<volume_dim>>,
+      analytic_compute, error_compute>;
```

Comment:
This relatively complicated code above is repeated in all subcell executables, as is much of the code in the `factory_creation` struct below. How about a `DgSubcell/Events/Factor.hpp` that builds the a list of `evolution::dg::subcell::Events::field_observations`, similar to the one in `::dg::Events::field_observations` but with all the `use_dg_subcell` switches?

Reply:
I'm unsure the code repetition bothers me enough to warrant the annoyance of yet another type alias that makes a bunch of internal decisions for me. The `factory_creation` is removed in the next commit already

Reply:
Up to you. What do you mean is removed in the next commit?

Reply:
Whoops, looks like it's not, but it can be, I think. The `tmpl::conditional_t` inside the `factory_creation` below